### ISSUE TITLE
Permission denied when running keepmenu

### ISSUE THREAD

**Issue by vivekkalyan** [firecat53/keepmenu#85]
OS: `5.11.11-arch1-1 #1 SMP PREEMPT Tue, 30 Mar 2021 14:10:17 +0000 x86_64 GNU/Linux`
keepmenu: `python-keepmenu-git r154.354c094-1`
pykeepass: `python-pykeepass 4.0.0-1`

```
Process DmenuRunner-2:
Traceback (most recent call last):
  File "/usr/bin/keepmenu", line 1547, in <module>
    MANAGER = client()
  File "/usr/bin/keepmenu", line 1267, in client
    mgr.connect()
  File "/usr/lib/python3.9/multiprocessing/managers.py", line 522, in connect
    conn = Client(self._address, authkey=self._authkey)
  File "/usr/lib/python3.9/multiprocessing/connection.py", line 507, in Client
    c = SocketClient(address)
  File "/usr/lib/python3.9/multiprocessing/connection.py", line 635, in SocketClient
    s.connect(address)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/bin/keepmenu", line 1318, in run
    self.dmenu_run()
  File "/usr/bin/keepmenu", line 1385, in dmenu_run
    sel = view_all_entries(list(options), filtered_entries)
  File "/usr/bin/keepmenu", line 911, in view_all_entries
    return dmenu_select(min(DMENU_LEN, len(options) + len(kp_entries)), inp=entries_b)
  File "/usr/bin/keepmenu", line 248, in dmenu_select
    Popen(cmd[0], stdin=PIPE, stdout=PIPE, env=ENV).communicate(input=err)
  File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: ''
```

Tried changing Permissions of the database to `chmod 666` but does not help. FWIW keepmenu is installed in `usr/bin/keepmenu` as root.

**Comment by firecat53** [comment]
1. How did you install keepmenu?
2. Are you running keepmenu as root or as your user?
3. Is this error from the initial run of keepmenu? 
4. Do your ~/.config/keepmenu and ~/.cache directories have the right permissions?

Try uninstalling keepmenu and installing (as your user) with `pip install --user keepmenu`

**Comment by asmallcarrot** [comment]
I have the same issue and haven't been able to find a fix. I can't run it as sudo because I can't find where to set the database file for the sudo keepmenu.

**Comment by firecat53** [comment]
It probably shouldn't be run as root anyways! Can you also answer 1, 3 and 4 above? Did you also try reinstalling with `pip install --user keepmenu` ?

**Comment by notnoxii** [comment]
Same issue, I installed originally from the AUR, but when removed and reinstalled through pip same issue. It was working for a day after the first install. both directories have r/w permissions.

**Comment by notnoxii** [comment]
Just fixed it by removing ~/.config/keepmenu and removing the AUR version, and installing PIP again.

**Comment by asmallcarrot** [comment]
I uninstalled keepmenu, removed the config file, tried every single version from aur to using the source. I made sure all file permissions were correct. And it still won't work.

**Comment by firecat53** [comment]
What window manager/dektop environment are you using?

**Comment by asmallcarrot** [comment]
DWM, I'm also using dmenu

**Comment by firecat53** [comment]
Any progress on this? I am unable to reproduce the failure at all. I'm assuming
you're using Arch. If you can give me an exact sequence of steps you are using
to install, maybe I can reproduce, but I've run Keepmenu on Arch for years with
no troubles. I feel like there's something funky with your system, but that's
the developer in me :)

**Comment by vivekkalyan** [comment]
Hey @firecat53 sorry for not keeping track of this convo. Eventually I manged to find the cause of the error, which is a patch I was using with dmenu ([case-insensitive](https://tools.suckless.org/dmenu/patches/case-insensitive/dmenu-caseinsensitive-5.0.diff)) which was causing the problems. The patch messed around with the accepted command line args for dmenu which caused keepmenu to fail. I have since removed that patch and don't expect keepmenu to work when the other patch changes the command line invocations.

@asmallcarrot Maybe you are facing the same problem?

Feel free to close this issue!
